PublicFolders/ValidateEXOPFDumpster.ps1

```diff
@@ -0,0 +1,423 @@
+    if ($PublicFolderInfo.Publicfolder.MailEnabled -eq $true) {
+        $Hasvalue=$PublicFolderInfo.Publicfolder.MailRecipientGuid.Guid
+        if ($null -eq $Hasvalue -or $Hasvalue -eq "00000000-0000-0000-0000-000000000000") {
+            $Fix= "FIX --> Please follow the following article https://aka.ms/EnableMPF to mail enable the affected public folder to generate a GUID over MailRecipientGuid parameter, validate MailRecipientGuid parameter has a GUID using https://aka.ms/getpf article then mail disable the affected public folder back again using https://aka.ms/disablempf article to be able to remove the public folder as requested."
+            $Issue="Mail-enabled public folder $($PublicFolderInfo.Publicfolder.Identity) is unhealthy e.g MailRecipientGuid parameter is found empty/null which is a blocker for deletion operations over the public folder"
+            WritetoScreenANDlogDlog -Issue $Issue -Fix $Fix
+        }
+    }
+}
+Function ExtractLog {
+    param([PSCustomObject]$PublicFolderInfo)
+    #create zip file for logs folder
+    $tstamp= Get-Date -Format yyyyMMdd_HHmmss
+    if (!(Test-Path  "$ExportPath\logs_$tstamp")) {
+        mkdir "$ExportPath\logs_$tstamp" -Force | Out-Null
+    }
+    $PublicFolderInfo.Publicfolder | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\Publicfolder.txt" -NoClobber
+    $PublicFolderInfo.Publicfolderdumpster | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\Publicfolderdumpster.txt" -NoClobber
+    $PublicFolderInfo.pfmbx | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\pfmbx.txt" -NoClobber
+    $PublicFolderInfo.PfMBXstats | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\PfMBXstats.txt" -NoClobber
+    $PublicFolderInfo.IPM_SUBTREE | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\IPM_SUBTREE.txt" -NoClobber
+    $PublicFolderInfo.NON_IPM_SUBTREE | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\NON_IPM_SUBTREE.txt" -NoClobber
+    $PublicFolderInfo.DUMPSTER_ROOT | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\DUMPSTER_ROOT.txt" -NoClobber
+    Compress-Archive -Path "$ExportPath\logs_$tstamp" -DestinationPath $ExportPath\logs_$tstamp
+}
+#Intro
+$ts = Get-Date -Format yyyyMMdd_HHmmss
+Write-Host $ExportPath
+If ($null -eq $ExportPath -or $ExportPath -eq "") {
+    $ExportPath = "$env:USERPROFILE\Desktop\ValidatePFDumpster\ValidatePFDumpster_$ts"
+    mkdir $ExportPath -Force | Out-Null
+} else {
+    if (!(Test-Path -Path $ExportPath)) {
+        Write-Host "The specified folder location $ExportPath is not existing, please re-run the script and ensure to enter a valid path or leave the ExportPath unassigned" -ForegroundColor Red
+        exit
+    } else {
+        $ExportPath = "$ExportPath\ValidatePFDumpster\ValidatePFDumpster_$ts"
+        mkdir $ExportPath -Force | Out-Null
+    }
+}
+$Script:Conditionsfailed = 0
+$Pfolder=Read-Host "Please enter affected public folder EntryID or Identity: e.g \PF1"
```

Comment:
No validation that this is a good public folder result.

Reply:
Yes as am listing only blockers on the report

Reply:
This should be a parameter rather than a prompt.